core/test/config/property_tree.cpp

```diff
@@ -181,3 +181,134 @@ TEST(PropertyTree, UseInCondition)
     ASSERT_EQ(first, 2);
     ASSERT_EQ(second, 1);
 }
+
+
+class PropertyTreeEquality : public ::testing::Test {
+protected:
+    PropertyTreeEquality()
+    {
+        auto generator = [](auto&& vector) {
+            // empty node
+            vector.emplace_back(pnode());
+            // boolean
+            vector.emplace_back(pnode(true));
+            // real
+            vector.emplace_back(pnode(1.2));
+            // integer
+            vector.emplace_back(pnode(4));
+            // string
+            vector.emplace_back(pnode("123"));
+            // array
+            vector.emplace_back(
+                pnode(pnode::array_type{pnode{"1"}, pnode{4}, pnode{true}}));
+            // array2
+            vector.emplace_back(
+                pnode(pnode::array_type{pnode{"2"}, pnode{4}, pnode{true}}));
+            // array3
+            vector.emplace_back(
+                pnode(pnode::array_type{pnode{"3"}, pnode{4}, pnode{true}}));
+            // map
```

Comment:
maybe have one of these maps also store a map (and maybe an array?). I think this should make the test `CheckEqualityUnderlyingNodeDiff` unnecessary.

Reply:
Yes, will it be a bit confusing or hidden?
When I add these different map and different array, I feel CheckEquality else-branch for different tag and different content